### PR TITLE
Snap thumb on custom ticks and add Ticks property on slider

### LINKED ISSUE #4192: Slider thumb doesn't snap on Ticks property from TickBar if set
The [existing logic](https://github.com/AvaloniaUI/Avalonia/blob/535c053338cea3177f1b7ba4b8ed1863516ba07d/src/Avalonia.Controls/Slider.cs#L240) for snapping the ticks on the Ticks property from TickBar is completely ignored, we should either just take the approach Microsoft did in [their own implementation](https://referencesource.microsoft.com/#PresentationFramework/src/Framework/System/windows/Controls/Slider.cs,1099) of WPF Sliders, either a new approach.

### PR DESCRIPTION
**Pull request by Mehigh17**
## What does the pull request do?
This pull requests ports the behavior of custom Ticks from the WPF Sliders to Avalonia Sliders. 

## What is the current behavior?
Currently when the TickBar has custom ticks set and the snap to tick is enabled, the thumb wont snap on them. Also in order to set Ticks on a Slider you must create a template and set the ticks on the TickBar, which is tedious.

## What is the updated/expected behavior with this PR?
A control has been created in the controls project where you can see the thumb snap to the custom created ticks, also ticks are set on the Slider in the code behind.

## How was the solution implemented (if it's not obvious)?
It's a port from WPF, the exact source of `SnapToTick` implementation is [here](https://referencesource.microsoft.com/#PresentationFramework/src/Framework/System/windows/Controls/Slider.cs,1099).

## Checklist

- [ ] Added unit tests (if possible)?
- [X] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Breaking changes
There are none.

## Fixed issues
Fixes #4192

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4219) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4219) before we can accept your contribution.<br/><br/>:x: Mehigh17 [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4219)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=4219) it.</sub>

**Review by jmacato — APPROVED** [review]
Nicely done! thanks alot @Mehigh17 :D

File: src/Avalonia.Controls/Slider.cs [hunk 1]
```diff
@@ -94,6 +102,15 @@ public Slider()
             UpdatePseudoClasses(Orientation);
         }
 
+        /// <summary>
+        /// Defines the ticks to be drawn on the tick bar.
+        /// </summary>
+        public List<double> Ticks
```
Comment (Gillibald) [inline src/Avalonia.Controls/Slider.cs:108]: I think making this a `AvaloniaList<double>` allows defining this in XAML `Ticks="0, 5, 20, 100"`

Comment (jmacato) [reply src/Avalonia.Controls/Slider.cs:108]: ah true... it should be avalonia list

Comment (Mehigh17) [reply src/Avalonia.Controls/Slider.cs:108]: The [Ticks property in TickBar](https://github.com/AvaloniaUI/Avalonia/blob/535c053338cea3177f1b7ba4b8ed1863516ba07d/src/Avalonia.Controls/TickBar.cs#L138) is also a `List<double>`, therefore we should change this first maybe.

Comment (jmacato) [reply src/Avalonia.Controls/Slider.cs:108]: @Mehigh17 perhaps you can change both in this PR?

Comment (Mehigh17) [reply src/Avalonia.Controls/Slider.cs]: @jmacato Fixed in [last commit](https://github.com/AvaloniaUI/Avalonia/pull/4219/commits/d8d9a1865dc4abe7466a43f14edbab7262222f7a).

**Review by jmacato — APPROVED** [review]

**Pull request merged**